pkg/services/accesscontrol/acimpl/service.go

```diff
@@ -243,8 +245,25 @@ func (s *Service) DeclarePluginRoles(ctx context.Context, ID, name string, regs
 }
 
 // SearchUsersPermissions returns all users' permissions filtered by action prefixes
-func (s *Service) SearchUsersPermissions(ctx context.Context, user identity.Requester,
+func (s *Service) SearchUsersPermissions(ctx context.Context, usr identity.Requester,
 	options accesscontrol.SearchOptions) (map[int64][]accesscontrol.Permission, error) {
+	if options.UserLogin != "" {
+		// Resolve userLogin -> userID
+		if err := options.ResolveUserLogin(ctx, s.userSvc); err != nil {
+			return nil, err
+		}
+		options.UserLogin = ""
+	}
+	if options.UserID > 0 {
+		// Reroute to the user specific implementation of search permissions
+		// because it leverages the user permission cache.
+		userPerms, err := s.SearchUserPermissions(ctx, usr.GetOrgID(), options)
+		if err != nil {
+			return nil, err
+		}
+		return map[int64][]accesscontrol.Permission{options.UserID: userPerms}, nil
+	}
+
 	timer := prometheus.NewTimer(metrics.MAccessSearchPermissionsSummary)
```

Comment:
Seems like observation happens after first return, so in some cases we do not measure it.

Reply:
Yes, that's intended. The `SearchUserPermission` function has another metric.